code/modules/mining/boulder_processing/boulder.dm

```diff
@@ -91,16 +103,28 @@
 		return FALSE
 
 /obj/item/boulder/attackby_secondary(obj/item/weapon, mob/user, params)
+	to_chat(world, "start")
 	. = ..()
+	to_chat(world, "pre-parent")
 	if(.)
+		to_chat(world, "failed parent")
 		return
-	if(!isliving(user))
-		return SECONDARY_ATTACK_CONTINUE_CHAIN
+	if(HAS_TRAIT(user, TRAIT_BOULDER_BREAKER) || HAS_TRAIT(weapon, TRAIT_BOULDER_BREAKER))
+		manual_process(weapon, user, INATE_SPEED_MULTIPLIER)
+		to_chat(world, "boulder breaker shrug")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 	if(weapon.tool_behaviour == TOOL_MINING)
 		manual_process(weapon, user)
-		return TRUE
-	if(isgolem(user))
-		manual_process(weapon, user, override_speed_multiplier = GOLEM_BREAK_SPEED_MULTIPLIER)
+		to_chat(world, "tool breaker")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	to_chat(world, "ended, failed")
```

Comment:
This won't get executed because just before this line you are returning.

I noticed in a lot of places you are trying to debug with to_chat() and say() which is definitely not ideal.

It is true that vs code debugger has stopped working after the 515 update which is why I would advise you to do https://github.com/tgstation/tgstation/issues/80008#issuecomment-1831471550

It will take a while before the next update comes in

Reply:
Please be aware that I've drafted the PR as I'm still working through a few code improvements and may not be available at the second. I would refrain from reviewing anything while I'm working on other reviews as well as yours, and I will undraft when it is stable enough to review. Apologies in advance.